record/record.go

```diff
@@ -169,9 +179,10 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 				}
 			}
 			r.last = chunkType == fullChunkType || chunkType == lastChunkType
+			r.recovering = false
 			return nil
 		}
-		if r.n < blockSize && r.started {
+		if r.n < blockSize && r.started && !r.recovering {
```

Comment:
Is this the right formula, or should it be:

if r.n < blockSize && r.started {
  if r.j != r.n && !r.recovering {
    return io.ErrUnexpectedEOF
  }
  return io.EOF
}